File: backend/MiniOrder.Domain/Entities/Order.cs
namespace MiniOrder.Domain.Entities;

public class Order
{
    public int Id { get; set; }

    public string CustomerName { get; set; } = string.Empty;

    public DateTimeOffset CreatedAt { get; set; }

    public decimal TotalAmount { get; set; }

    public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
}

File: frontend/src/types/order.ts
export interface CreateOrderItemRequest {
  productId: number;
  quantity: number;
}

export interface CreateOrderRequest {
  customerName: string;
  items: CreateOrderItemRequest[];
}

export interface OrderItem {
  productId: number;
  productName: string;
  stockCode: string;
  quantity: number;
  unitPrice: number;
  lineTotal: number;
}

export interface Order {
  id: number;
  customerName: string;
  createdAt: string;
  totalAmount: number;
  items: OrderItem[];
}
